hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/v2/components/navBar/navBar.tsx

```diff
@@ -0,0 +1,182 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useState, useEffect } from 'react';
+import axios, { AxiosResponse } from 'axios';
+import { Layout, Menu, Spin } from 'antd';
+import {
+  BarChartOutlined,
+  ClusterOutlined,
+  ContainerOutlined,
+  DashboardOutlined,
+  DatabaseOutlined,
+  DeploymentUnitOutlined,
+  FolderOpenOutlined,
+  InboxOutlined,
+  LayoutOutlined,
+  PieChartOutlined
+} from '@ant-design/icons';
+import { useLocation, Link } from 'react-router-dom';
+
+
+import logo from '@/logo.png';
+import { showDataFetchError } from '@/utils/common';
+import { AxiosGetHelper, cancelRequests } from '@/utils/axiosRequestHelper';
+
+import './navBar.less';
+
+
+// ------------- Types -------------- //
+type NavBarProps = {
+  collapsed: boolean;
+  onCollapse: (arg0: boolean) => void;
+}
+
+const NavBar: React.FC<NavBarProps> = ({
+  collapsed = false,
+  onCollapse = () => { }
+}) => {
+  const [isHeatmapEnabled, setIsHeatmapEnabled] = useState<boolean>(false);
+  let cancelDisabledFeatureSignal: AbortController;
+  const location = useLocation();
+
+  const fetchDisabledFeatures = async () => {
+    const disabledfeaturesEndpoint = `/api/v1/features/disabledFeatures`;
+    const { request, controller } = AxiosGetHelper(
+      disabledfeaturesEndpoint,
+      cancelDisabledFeatureSignal
+    )
+    cancelDisabledFeatureSignal = controller;
+    try {
+      const response: AxiosResponse<string[]> = await request;
+      const heatmapDisabled = response?.data?.includes('HEATMAP')
+      setIsHeatmapEnabled(!heatmapDisabled)
+    } catch (error: unknown) {
+      showDataFetchError((error as Error).toString())
+    }
+  }
+
+
+  useEffect(() => {
+    fetchDisabledFeatures();
+    // Component will unmount
+    return (() => {
+      cancelRequests([cancelDisabledFeatureSignal])
+    })
+  }, [])
```

Comment:
```suggestion
  }, [fetchDisabledFeatures])
```

Reply:
I think this won't be required, because in case the Heatmap is disabled on the first fetch/page load, we are sure it won't be re-enabled while using Recon. Hence we do not need to track the feature.

Also since the fetching of the requests is done only on the initial load, we are sure fetchDisabledFeatures will always be the same hence this will never be triggered

Reply:
> I think this won't be required, because in case the Heatmap is disabled on the first fetch/page load, we are sure it won't be re-enabled while using Recon. Hence we do not need to track the feature.
> 
> Also since the fetching of the requests is done only on the initial load, we are sure fetchDisabledFeatures will always be the same hence this will never be triggered on subsequent renders.

I am not sure if this is correct, because though initial load may fetch that heatmap is disabled, but then user can enable it and expect that to display, so then  we should correct this. Am I missing something here ?

Reply:
But won't the user need to set the configuration in the xml file and restart the service?

Reply:
> But won't the user need to set the configuration in the xml file and restart the service?

Yes restart will happen, so if user is on Datanodes page and just refresh that page after enabling heatmap flag,  will the code call fetchDIsabledFeatures ?, if not, then how the heatmap information whether enabled or disabled will get on Datanodes page refresh ? Expectation is to start showing heatmap nav item , even if user refreshes just Datanodes page. Pls check this part.